symphony-bdk-core/src/main/java/com/symphony/bdk/core/auth/impl/AuthenticationRetry.java

```diff
@@ -0,0 +1,88 @@
+package com.symphony.bdk.core.auth.impl;
+
+import com.symphony.bdk.core.auth.exception.AuthUnauthorizedException;
+import com.symphony.bdk.core.config.model.BdkRetryConfig;
+import com.symphony.bdk.core.retry.RetryWithRecovery;
+import com.symphony.bdk.core.retry.RetryWithRecoveryBuilder;
+import com.symphony.bdk.core.util.function.SupplierWithApiException;
+import com.symphony.bdk.http.api.ApiException;
+import com.symphony.bdk.http.api.ApiRuntimeException;
+
+import org.apiguardian.api.API;
+
+import javax.ws.rs.ProcessingException;
+
+/**
+ * Class used to implement the specific logic for authentication calls.
+ * Delegates the retry mechanism to {@link RetryWithRecovery}.
+ * @param <T> the type returned by the authentication call.
+ */
+@API(status = API.Status.INTERNAL)
+class AuthenticationRetry<T> {
+
+  private RetryWithRecoveryBuilder<T> baseRetryBuilder;
+
+  /**
+   * Creates a {@link RetryWithRecoveryBuilder} with the retry condition specific to authentication calls.
+   *
+   * @param retryConfig the retry configuration to be used.
+   * @param <T> the type returned by the authentication call.
+   * @return a {@link RetryWithRecoveryBuilder} instance initialized with the provided retry configuration
+   * and the retry condition {@link #canAuthenticationBeRetried(Throwable)}.
+   */
+  public static <T> RetryWithRecoveryBuilder<T> getBaseRetryBuilder(BdkRetryConfig retryConfig) {
```

Comment:
I am not sure why you have to make this method `public static`. I think we can easily create an `AuthenticationRetry` instance and get the `RetryWithRecoveryBuilder` by a getter?

Reply:
This is used in the `AbstractExtensionAppAuthenticator` when fetching the pod certificate. For this one, I thought we wanted to have the same `RetryWithRecovery` as for the authentication (do not authenticate again in case of 401) but use `RetryWithRecovery.executeAndRetry`. To have a consistent design, maybe we could maybe have everything static?